### PR TITLE
Add Terminology documentation

### PR DESCRIPTION
**Pull request by jnv**
This is a draft.

- Are you okay with the overall structure of the document?
- Are there any terms missing?
- Is anything else missing?

---
This change is [Reviewable](https://reviewable.io/reviews/cvut/sirius/187)

### CODE REVIEW

File: docs/terminology.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+:kTeacher: link:{:kosapi:}Teacher[Teacher (Vyučující)]
+:kTimetableSlot: link:{:kosapi:}Parallel#TimetableSlot[TimetableSlot (Rozvrhový lístek)]
+:kTeacherTimetableSlot: link:{:kosapi:}Teachers#GET-teachersusernameOrIdtimetable[TeacherTimetableSlot (Omezení vyučujícího)]
+
+Explanation of terms used throughout Sirius code-base, along with Czech translation and equivalent concepts in KOSapi.
+
+Course (Předmět)::
+Corresponds to KOSapi's {kCourse}.
+Event (Událost)::
+Universal event type. Can be either imported directly from KOSapi's {kCourseEvent} and {kExam}, or generated, or generated from TimetableSlot.
```
Comment (flexik) [inline docs/terminology.adoc]: Duplicit `or generated`.

File: docs/terminology.adoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Faculty-Semester (Fakulto-Semestr)::
+A representation of Semester bound to a specific faculty. Composes of multiple Semester Periods.
+Parallel (Paralelka)::
+Corresponds to KOSapi's {kParallel}.
+Person (Osoba)::
+Corresponds to KOSapi's {kPerson}.
+Room (Místnost)::
+Corresponds to KOSapi's {kRoom}.
+Semester Period (Perioda semestru)::
```
Comment (flexik) [inline docs/terminology.adoc]: What about a different translation? `Perioda` in Czech refers to something that repeats, but `Semester period` or `Faculty semester` are not recurrent. My suggestions are either `Období semestru` or `Část semestru`.

File: docs/terminology.adoc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* holiday period.
++
+Schedule Exception (Rozvrhová výjimka)::
+Exception of regular scheduling event; see <<schedule-exceptions.adoc, Schedule exceptions>>.
+Teacher (Vyučující)::
+A person which is registered as Teacher in event or who has TeacherTimetableSlot.
+Timetable Slot (Rozvrhový lístek)::
+Within Parallel, corresponds to KOSapi's {kTimetableSlot}.
+Teacher Timetable Slot (Omezení vyučujícího)::
+Information about teacher's regular availability. Corresponds to KOSapi's {kTeacherTimetableSlot}.
```
Comment (flexik) [inline docs/terminology.adoc]: `unavailability` would be probably better here. Teacher timetable slots can of course be used to show availability, but most of the time they are used in opposite manner.

**Comment by jirutka** [comment]
Merged in a20a851ca8b1165e5a63d97da2ba9133c45bdccd, thanks!

**Pull request closed without merging**